src/dct/systems/bldgPersist.lua

```diff
@@ -9,25 +9,30 @@ local Marshallable = require("dct.libs.Marshallable")
 local Logger = require("dct.libs.Logger").getByName("System")
 
 local SceneryTracker = class("SceneryTracker", Marshallable)
-function SceneryTracker:__init(theater)
+function SceneryTracker:__init()
 	Marshallable.__init(self)
 	self.destroyed = {}
-	theater:addObserver(self.onDCSEvent, self,
-		self.__clsname..".onDCSEvent")
 	self:_addMarshalNames({
 		"destroyed",
 	})
 end
 
+function SceneryTracker:postinit(theater)
+	theater:addObserver(self.onDCSEvent, self,
+		self.__clsname..".onDCSEvent")
+end
+
 function SceneryTracker:_unmarshalpost()
-	for _, bldg in pairs(self.destroyed) do
-		local id = tonumber(bldg)
-		local obj = {id_ = id}
-		local pt = Object.getPoint(obj)
-		trigger.action.explosion(pt, 500)
+	for bldg, _ in pairs(self.destroyed) do
+		local pt = Object.getPoint({id_ = tonumber(bldg)})
+		trigger.action.explosion(pt, 250)
```

Comment:
250 is a large explosion size, likely overkill for removing a building. I believe the number is essentially the defined number of KGs of TNT you want to simulate - in this case 250kgs of TNT. I'll do some testing and find out what an appropriate value will be.

Reply:
250 is basically a 1000lb bomb. It was 1000 before. We can change it to a 500lb (87-90kg).